src/Window.res

```diff
@@ -155,3 +178,56 @@ let isSandbox = Location.hostname === "beta.hyperswitch.io"
 let isInteg = Location.hostname === "dev.hyperswitch.io"
 
 let isProd = Location.hostname === "checkout.hyperswitch.io"
+
+let isIframed = () =>
+  try {
+    Location.href !== Top.Location.href
+  } catch {
+  | e => {
+      let default = true
+      Js.Console.error3(
+        "Failed to check whether or not document is within an iframe",
+        e,
+        `Using "${default->String.make}" as default (due to DOMException)`,
+      )
+      default
+    }
+  }
+
+let getRootHostName = () =>
+  switch isIframed() {
+  | true =>
+    try {
+      Top.Location.hostname
+    } catch {
+    | e => {
+        let default = Location.hostname
+        Js.Console.error3(
+          "Failed to get root document's hostname",
+          e,
+          `Using "${default}" [window.location.hostname] as default`,
+        )
+        default
+      }
+    }
+  | false => Location.hostname
+  }
+
+let replaceRootHref = (href: string) => {
+  switch isIframed() {
+  | true =>
+    try {
+      Top.Location.replace(href)
+    } catch {
+    | e => {
+        Js.Console.error3(
```

Comment:
@kashif-m why would this crash? You should always be able to call `window.top.location.replace()` right?

Reply:
If the iframe is sandboxed and does not have `allow-top-navigation` permission, it will throw DOMException

<img width="645" alt="Screenshot 2024-09-09 at 2 52 02 PM" src="https://github.com/user-attachments/assets/d3725645-ce26-4519-9102-1067bbbb14a3">

Reply:
Ok. I wasn't aware of `allow-top-navigation` permission. Makes sense.